htmlreport/info_from_cmd.py

```diff
@@ -0,0 +1,52 @@
+import subprocess
+import os
+import datetime
+import locale
+
+
+def git_blame(line, path, file, blame_options):
+    git_blame_dict = {
+        'author': 'Unknown',
+        'author_mail': '---',
+        'date': '---'
+    }
+    head, tail = os.path.split(file)
+    if head != "":
+        path = head
+
+    try:
+        os.chdir(path)
+    except:
+        return git_blame_dict
+
+    try:
+        result = subprocess.check_output('git blame -L %d%s%s --porcelain -- %s' % (line, " -w" if "-w" in blame_options else "", " -M" if "-M" in blame_options else "", file))
+        result = result.decode(locale.getpreferredencoding())
+    except:
+        return git_blame_dict
+
+    if result.startswith('fatal'):
+        return git_blame_dict
+
+    author = result[result.find('author') + 6:result.find('author-mail') - 1]
```

Comment:
the result is:
```
4710d80a4087192d06829e1839208feba383c5c3 101 100 1
author IOBYTE
author-mail <reif@earthlink.net>
author-time 1516783882
author-tz -0500
committer amai2012
committer-mail <amai2012@users.noreply.github.com>
committer-time 1516783882
committer-tz +0100
summary Fix #5766 (FP: typedef array throws off parser) (#1052)
....
```
I think we should split this up more intelligently. Something like this maybe:

```
for line in results.split('\n')[1:]:
    space_pos = line.find(' ')
    if space_pos > 30:
        break
    key = line[:space_pos]
    val = line[space_pos + 1:]
    git_blame_dict[key] = val
```

.. I did not test that but the intention is to extract all the fields for the first SHA into the dict.

Reply:
Yes, it makes sense. I have changed it and now it makes sense and it is more reliable. Thanks